From 2df77f845102bc493fa9537e7c5df2640eda68a4 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Tue, 23 Jun 2020 19:45:29 +0200
Subject: [PATCH 028/351] ARM: dts: sun8i-a83t-tbs-a711: Add support for the
 front camera (gc2145)

The tablet has two cameras, that can be switched between. Add
support for the front one.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 arch/arm/boot/dts/sun8i-a83t-tbs-a711.dts | 41 +++++++++++++++++++++--
 1 file changed, 39 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/sun8i-a83t-tbs-a711.dts b/arch/arm/boot/dts/sun8i-a83t-tbs-a711.dts
index af9a83988e5e..b06dd5d3dac0 100644
--- a/arch/arm/boot/dts/sun8i-a83t-tbs-a711.dts
+++ b/arch/arm/boot/dts/sun8i-a83t-tbs-a711.dts
@@ -252,7 +252,11 @@ &csi {
 };
 
 &csi_in {
-	csi_hm5065_ep: endpoint {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	csi_hm5065_ep: endpoint@0 {
+		reg = <0>;
 		remote-endpoint = <&hm5065_ep>;
 		bus-width = <8>;
 		data-active = <1>;
@@ -260,6 +264,16 @@ csi_hm5065_ep: endpoint {
 		hsync-active = <0>;
 		vsync-active = <1>;
 	};
+
+	csi_gc2145_ep: endpoint@1 {
+		reg = <1>;
+		remote-endpoint = <&gc2145_ep>;
+		bus-width = <8>;
+		hsync-active = <1>;
+		vsync-active = <1>;
+		data-active = <1>;
+		pclk-sample = <1>;
+	};
 };
 
 &de {
@@ -323,7 +337,7 @@ npc100: nfc@28 {
 };
 
 &i2c_gpio {
-	hm5065: camera@1f {
+	hm5065: rear-camera@1f {
 		compatible = "himax,hm5065";
 		reg = <0x1f>;
 		clocks = <&ccu CLK_CSI_MCLK>;
@@ -346,6 +360,29 @@ hm5065_ep: endpoint {
 			};
 		};
 	};
+
+	gc2145: front-camera@3c {
+		compatible = "galaxycore,gc2145";
+		reg = <0x3c>;
+		clocks = <&ccu CLK_CSI_MCLK>;
+		clock-names = "xclk";
+		IOVDD-supply = <&reg_dldo3>;
+		AVDD-supply = <&reg_dldo4>;
+		DVDD-supply = <&reg_eldo3>;
+		reset-gpios = <&pio 4 16 (GPIO_ACTIVE_LOW | GPIO_OPEN_DRAIN)>; /* PE16 */
+		enable-gpios = <&pio 4 17 (GPIO_ACTIVE_LOW | GPIO_OPEN_DRAIN)>; /* PE17 */
+
+		port {
+			gc2145_ep: endpoint {
+				remote-endpoint = <&csi_gc2145_ep>;
+				bus-width = <8>;
+				hsync-active = <1>;
+				vsync-active = <1>;
+				data-active = <1>;
+				pclk-sample = <1>;
+			};
+		};
+	};
 };
 
 &i2s0 {
-- 
2.34.0

